src/stakers/stakers.controller.ts

```diff
@@ -22,6 +23,46 @@ import { StakersService } from './stakers.service';
 export class StakersController {
   constructor(private readonly stakersService: StakersService) {}
 
+
+  @Get('/ns/all')
+  @ApiOperation({
+    summary: 'Fetch Native Stake balances for a specific date interval',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'The records were successfully fetched.',
+    type: NativeStakeBalanceDto,
+  })
+  @UseInterceptors(HttpDateCacheInterceptor)
+  @CacheTTL(60e3)
+  async getAllNativeStakes(
+    @Query() query: SnapshotsIntervalDto,
+  ): Promise<NativeStakeBalancesDto> {
+    if (query.startDate && query.endDate) {
+      if (Date.parse(query.startDate) > Date.parse(query.endDate)) {
+        throw new HttpException(
+          'startDate is later than endDate',
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+    }
+    if (!query.startDate && !query.endDate) {
+      throw new HttpException(
+        'No startDate or endDate provided',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
```

Comment:
(non-blocking/optional comment) This is an exact copy-paste of the validation in `getNativeStakes` below, I'd consider wrapping it in a helper function which accepts the `query` as a parameter and performs the checks.